google-cloud-spanner-hibernate-dialect/src/main/java/com/google/cloud/spanner/hibernate/SpannerDialect.java

```diff
@@ -256,4 +259,48 @@ public SqmMultiTableInsertStrategy getFallbackSqmInsertStrategy(
       EntityMappingType entityDescriptor, RuntimeModelCreationContext runtimeModelCreationContext) {
     return NoOpSqmMultiTableInsertStrategy.INSTANCE;
   }
+
+  @Override
+  public String addSqlHintOrComment(
+      String sql,
+      QueryOptions queryOptions,
+      boolean commentsEnabled) {
+    if (hasCommentHint(queryOptions)) {
+      sql = applyHint(sql, queryOptions.getComment());
+    }
+    if (queryOptions.getDatabaseHints() != null && !queryOptions.getDatabaseHints().isEmpty()) {
+      sql = applyQueryHints(sql, queryOptions);
+    }
+    return super.addSqlHintOrComment(sql, queryOptions, commentsEnabled);
+  }
+
+  private String applyHint(String sql, String hint) {
+    try {
+      return ReplaceQueryPartsHint.fromComment(hint).replace(sql);
+    } catch (Throwable ignore) {
+      // Just ignore and continue with the query normally.
```

Comment:
Why? This way user would not know about the issue. Should we atleast log the error?

Reply:
The reason that we ignore it here is that we don't actually know that it is a hint. All we know is that:
1. Something in the query generation chain has added a  comment to the query.
2. That comment contains some substrings that we expect when it is a hint.
3. BUT: It could in theory also be a coincidence that it contains those keywords, and just a comment that should not be treated as a hint. (We could make the check for substrings that must be in the comment for it to be considered a hint even more strict, but that would change the behavior; hints that for example contain a typo would then not pass that stricter check, and also not cause an error)

I've added a log statement for it.